external/llvm-project/mlir/lib/Conversion/AMDGPUToROCDL/AMDGPUToROCDL.cpp

```diff
@@ -259,21 +263,34 @@ struct RawBufferOpLowering : public ConvertOpToLLVMPattern<GpuOp> {
 };
 
 struct LDSBarrierOpLowering : public ConvertOpToLLVMPattern<LDSBarrierOp> {
-  using ConvertOpToLLVMPattern<LDSBarrierOp>::ConvertOpToLLVMPattern;
+  LDSBarrierOpLowering(LLVMTypeConverter &converter, Chipset chipset)
```

Comment:
We tried this before but there is a long standing performance issue regarding this.
I don't think this has been fixed yet.

Reply:
Wasn't that perf issue to do with `vmmcnt(0) & lgkmcnt(0)`?

Reply:
But yeah, I can perf this

Reply:
I looked, and it doesn't seem to be there. I can run perf to confirm, but still.

Reply:
I'll note this isn't the barrier with a fence, this is the intrinsics that specifically cause s_waitcnt and s_barrier

Reply:
It used to happens on the first LDS wait in the main loop of the xdlops gemm.
The loop starts with a global load which is independent to the LDS data, deliberately try to overlap global/LDS latency.

Reply:
Ok, yeah, the weirdness came back, I did check that exact spot. I'm going to go while an the compiler team